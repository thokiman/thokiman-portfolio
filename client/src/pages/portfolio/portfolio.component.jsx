import React, { lazy, Suspense } from "react";
import { Route } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import PageShell from "../../components/page-shell/page-shell.component";
import { selectIsSideBarHidden } from "../../redux/header/header.selectors";
import {
  selectAllItems,
  selectBrandItems,
  selectBrandRoute,
  selectDigitalArtItems,
  selectDigitalArtRoute,
  selectPhotographyBwItems,
  selectPhotographyBwRoute,
  selectPhotographyColorItems,
  selectPhotographyColorRoute,
  selectPortfolioRoute,
} from "../../redux/collection/collection.selectors";
import { PortfolioContainer, PortfolioSideBarHide } from "./portfolio.styles";
import SpinnerLoading from "../../components/spinner-loading/spinner-loading.component";
const PortfolioContentContainer = lazy(() =>
  import("../../components/portfolio-content/portfolio-content-container")
);
const PortfolioHeader = lazy(() =>
  import("../../components/portfolio-header/portfolio-header.component")
);

export const Portfolio = ({
  portfolioRoute,
  allItems,
  brandItems,
  brandRoute,
  digitalArtItems,
  digitalArtRoute,
  photographyBwItems,
  photographyBwRoute,
  photographyColorItems,
  photographyColorRoute,
  isSideBarHidden,
}) => {
  return (
    <PortfolioContainer $issidebarhidden={isSideBarHidden}>
      <Suspense fallback={<SpinnerLoading />}>
        {isSideBarHidden ? (
          <PortfolioSideBarHide>
            Port
            <br />
            folio
          </PortfolioSideBarHide>
        ) : null}
        <PortfolioHeader />
        <Route
          exact
          path={portfolioRoute}
          render={() => <PortfolioContentContainer items={allItems} />}
        />
        <Route
          path={brandRoute}
          render={() => <PortfolioContentContainer items={brandItems} />}
        />
        <Route
          path={digitalArtRoute}
          render={() => <PortfolioContentContainer items={digitalArtItems} />}
        />
        <Route
          path={photographyBwRoute}
          render={() => (
            <PortfolioContentContainer items={photographyBwItems} />
          )}
        />
        <Route
          path={photographyColorRoute}
          render={() => (
            <PortfolioContentContainer items={photographyColorItems} />
          )}
        />
      </Suspense>
    </PortfolioContainer>
  );
};
const mapStateToProps = createStructuredSelector({
  portfolioRoute: selectPortfolioRoute,
  allItems: selectAllItems,
  brandItems: selectBrandItems,
  brandRoute: selectBrandRoute,
  digitalArtItems: selectDigitalArtItems,
  digitalArtRoute: selectDigitalArtRoute,
  photographyBwItems: selectPhotographyBwItems,
  photographyBwRoute: selectPhotographyBwRoute,
  photographyColorItems: selectPhotographyColorItems,
  photographyColorRoute: selectPhotographyColorRoute,
  isSideBarHidden: selectIsSideBarHidden,
});

export default connect(mapStateToProps)(PageShell(Portfolio));
