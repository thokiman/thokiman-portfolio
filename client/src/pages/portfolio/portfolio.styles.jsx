import styled, { css, keyframes } from 'styled-components';
const blackColor = '#000000';
const whiteSmokeColor = '#e3e1e4';
const maxWidthSmartphone = 800;

export const PortfolioElement = styled.div`
  &.portfolio-element {
  }
`;

const portfolioSideBarHide = keyframes`
  0% {
    transform: rotateX(100deg);
    transform-origin: bottom;
    opacity: 0;
  }
  100% {
    transform: rotateX(0);
    transform-origin: bottom;
    opacity: 1;
  }

`;

const getTopPortfolioIsNotActivatedProps = ({
  $portfolioisnotactivatedprops,
}) => {
  return $portfolioisnotactivatedprops.top;
};
const getLeftPortfolioIsNotActivatedProps = ({
  $portfolioisnotactivatedprops,
}) => {
  return $portfolioisnotactivatedprops.left;
};
const getFontSizePortfolioIsNotActivatedProps = ({
  $portfolioisnotactivatedprops,
}) => {
  return $portfolioisnotactivatedprops.fontSize;
};

export const PortfolioIsNotActivated = styled.div`
  font-family: spitfire;
  color: ${whiteSmokeColor};
  position: absolute;
  top: ${getTopPortfolioIsNotActivatedProps}vh;
  left: ${getLeftPortfolioIsNotActivatedProps}vw;
  font-size: ${getFontSizePortfolioIsNotActivatedProps}vw;
  letter-spacing: 0.17em;
  width: 50vw;
  line-height: 1.5em;
  animation: ${portfolioSideBarHide} 2s cubic-bezier(0.175, 0.885, 0.32, 1.275)
    both;
`;
const PortfolioIsActivated = css`
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    position: relative;
    left: 0;
    transition: left 0.5s ease;
    opacity: 1;
    width: 100vw;
  }
`;
const getPortfolioSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return null;
  return PortfolioIsActivated;
};

export const PortfolioContainer = styled.div`
  background-color: ${blackColor};

  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 64vw;
    position: relative;
    top: 0;
    left: 140px;
    right: 0;
    bottom: 0;
    transition: left 0.5s ease;
    height: 130vh;
    overflow-y: scroll;
    overflow-x: hidden;
    opacity: 0.5;
  }
  ${getPortfolioSideBarActive};
`;
PortfolioContainer.displayName = 'PortfolioContainer';
