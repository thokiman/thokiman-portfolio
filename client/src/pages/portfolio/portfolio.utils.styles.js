import {
  hasScreenSize1,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize4,
  hasScreenSize5,
} from 'utils/common.component.utils';

export const measurePortfolioIsNotActivated = (viewWidth, viewHeight) => {
  let portfolioIsNotActivatedProps;

  if (hasScreenSize1(viewWidth)) {
    console.log(1);
    portfolioIsNotActivatedProps = {
      top: 23.5,
      left: 6,
      fontSize: 7,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log(2);
    portfolioIsNotActivatedProps = {
      top: 32,
      left: 6,
      fontSize: 5,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log(3);
    portfolioIsNotActivatedProps = {
      top: 28.5,
      left: 6,
      fontSize: 8,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log(4);
    portfolioIsNotActivatedProps = {
      top: 29,
      left: 6,
      fontSize: 9,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log(5);
    portfolioIsNotActivatedProps = {
      top: 31,
      left: 6,
      fontSize: 7,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log(6);
    portfolioIsNotActivatedProps = {
      top: 33,
      left: 6,
      fontSize: 6,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log(7);
    portfolioIsNotActivatedProps = {
      top: 39,
      left: 6,
      fontSize: 5,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log(8);
    portfolioIsNotActivatedProps = {
      top: 33,
      left: 6,
      fontSize: 12,
    };
  } else {
    console.log(9);
    portfolioIsNotActivatedProps = {
      top: 27.5,
      left: 6,
      fontSize: 10,
    };
  }
  return portfolioIsNotActivatedProps;
};
