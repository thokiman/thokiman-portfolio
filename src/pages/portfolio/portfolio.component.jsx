import React from "react";
import { Route } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import PageShell from "../../components/page-shell/page-shell.component";
import PortfolioContainer from "../../components/portfolio-container/portfolio-container.component";
import PortfolioHeader from "../../components/portfolio-header/portfolio-header.component";

import {
  selectAllItems,
  selectBrandItems,
  selectBrandRoute,
  selectDigitalArtItems,
  selectDigitalArtRoute,
  selectPhotographyBwItems,
  selectPhotographyBwRoute,
  selectPhotographyColorItems,
  selectPhotographyColorRoute,
} from "../../redux/collection/collection.selectors";
import "./portfolio.styles.scss";

const Portfolio = ({
  match: { url },
  allItems,
  brandItems,
  brandRoute,
  digitalArtItems,
  digitalArtRoute,
  photographyBwItems,
  photographyBwRoute,
  photographyColorItems,
  photographyColorRoute,
}) => {
  return (
    <div className="collections-container">
      <PortfolioHeader />
      <Route
        exact
        path={`${url}`}
        render={() => <PortfolioContainer items={allItems} />}
      />
      <Route
        path={`${url}${brandRoute}`}
        render={() => <PortfolioContainer items={brandItems} />}
      />
      <Route
        path={`${url}${digitalArtRoute}`}
        render={() => <PortfolioContainer items={digitalArtItems} />}
      />
      <Route
        path={`${url}${photographyBwRoute}`}
        render={() => <PortfolioContainer items={photographyBwItems} />}
      />
      <Route
        path={`${url}${photographyColorRoute}`}
        render={() => <PortfolioContainer items={photographyColorItems} />}
      />
    </div>
  );
};
const mapStateToProps = createStructuredSelector({
  allItems: selectAllItems,
  brandItems: selectBrandItems,
  brandRoute: selectBrandRoute,
  digitalArtItems: selectDigitalArtItems,
  digitalArtRoute: selectDigitalArtRoute,
  photographyBwItems: selectPhotographyBwItems,
  photographyBwRoute: selectPhotographyBwRoute,
  photographyColorItems: selectPhotographyColorItems,
  photographyColorRoute: selectPhotographyColorRoute,
});

export default PageShell(connect(mapStateToProps)(Portfolio));
