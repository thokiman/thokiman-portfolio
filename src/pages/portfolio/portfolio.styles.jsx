import styled, { css } from "styled-components";

const PortfolioSideBarActive = css`
  @media only screen and (max-width: 800px) {
    position: relative;
    left: 0;
    transition: left 0.5s ease;
    opacity: 1;
    width: 100vw;
  }
`;
const getPortfolioSideBarActive = (props) => {
  if (props.$issidebarhidden) return null;
  return PortfolioSideBarActive;
};

export const PortfolioContainer = styled.div`
  background-color: black;
  @media only screen and (max-width: 800px) {
    width: 65vw;
    position: relative;
    top: 0;
    left: 140px;
    right: 0;
    bottom: 0;
    transition: left 0.5s ease;
    height: 130vh;
    overflow-y: scroll;
    overflow-x: hidden;
    opacity: 0.5;
  }
  ${getPortfolioSideBarActive};
`;
PortfolioContainer.displayName = "PortfolioContainer";
