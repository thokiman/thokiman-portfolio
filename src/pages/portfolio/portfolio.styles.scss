.collections-container{
    background-color: black;
    .options{
        display: block;
        background: url('../../assets/layout/water-2.jpg'),rgba(227, 225, 228,0.6);
        background-blend-mode:overlay;
        background-size: cover;
        height:10vh;
        display: flex;
        justify-content:flex-start;
        align-items: center;
         

        .option{
        margin: auto 73px;
        text-decoration: none;
        cursor: pointer;
        letter-spacing: 0.04em;
         font-size: 150%;
         padding: 10px 10px;
         color:#128194;
         text-align: center;
         font-family: futura-medium;
         position: relative;
         left: 215px;
        }
    }
    .image-collections-container{
        
        
        display: grid;
        grid-gap: 10px;
        grid-template-columns: repeat(3, 1fr);
        grid-template-rows: repeat(auto-fill,1fr);
        column-gap: 10px;
        
        
        .image-collection{
            
            width: 100%;
            height: 100%;
            &:nth-child(5n){
                grid-column-end: span 2;
            }
        
         .image-item{
             display: flex;
             
             object-fit: cover;
             height:100%;
             width:100%;
         }   
        }
    }
}


