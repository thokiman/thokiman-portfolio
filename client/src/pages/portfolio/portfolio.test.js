import React from "react";
import { shallow } from "enzyme";

import { Portfolio } from "./portfolio.component";

describe("Portfolio page", () => {
  let wrapper;
  const mockRoute = "/";
  const mockArray = [];
  const mockFalseBool = false;
  beforeEach(() => {
    const mockProps = {
      portfolioRoute: mockRoute,
      allItems: mockArray,
      brandItems: mockArray,
      brandRoute: mockRoute,
      digitalArtItems: mockArray,
      digitalArtRoute: mockRoute,
      photographyBwItems: mockArray,
      photographyBwRoute: mockRoute,
      photographyColorItems: mockArray,
      photographyColorRoute: mockRoute,
      isSideBarHidden: mockFalseBool,
    };
    wrapper = shallow(<Portfolio {...mockProps} />);
  });
  it("should render Portfolio page", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should not render PortfolioSideBarHide if isSideBarHidden is false", () => {
    expect(wrapper.exists("PortfolioSideBarHide")).toBe(false);
  });
});
