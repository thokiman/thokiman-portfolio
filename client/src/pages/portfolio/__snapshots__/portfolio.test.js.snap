// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio page should render Portfolio page 1`] = `ShallowWrapper {}`;
